internal/component/azure/servicebus/subscription.go

```diff
@@ -361,14 +359,11 @@ func (s *Subscription) RenewLocksBlocking(ctx context.Context, receiver Receiver
 				s.logger.Debugf("Renewed session %s locks for %s", sessionReceiver.SessionID(), s.entity)
 			} else {
 				// Snapshot the messages to try to renew locks for.
-				s.mu.RLock()
-				msgs := make([]*azservicebus.ReceivedMessage, 0, len(s.activeMessages))
-				var i int
-				for _, m := range s.activeMessages {
-					msgs[i] = m
-					i++
-				}
-				s.mu.RUnlock()
+				msgs := make([]*azservicebus.ReceivedMessage, 0)
```

Comment:
Should we initialize it with some initial default capacity? Just thinking so, as with initial capacity of 0, re-allocations of slice will definitely happen.

Reply:
Yh makes sense to have an initial buffer, does 64 sound reasonable?

Reply:
This slice is being sent to RenewLockMessage for ASB, I am not very sure that how many number of messages can usually be the minimal kind of value for it. So, yeah 64 sounds fine or may be if you think to lower the value to like 16 or so. @ItalyPaleAle @berndverst suggestions?

Reply:
How about rather than guessing, we initialize it with size 0. Then, if we need a bigger slice in the future (e.g. 1), we replace the slice with newly-allocated memory. Every time we encounter a situation where we need more memory, we replace the slice with a bigger one and keep the bigger one.